### PR TITLE
LVPYT-4 user registration

### PR DESCRIPTION
**Pull request by vova-kul**

### CODE REVIEW

File: itaplay/itaplay/authentication/forms.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    class Meta:
+        model = User
+        fields = ('first_name', 'last_name', 'password', "confirm_password")
+
+    def clean_confirm_password(self):
+        """
+        Function checks that fields password and confirm_password are equal
+        :return: nothing
+        """
+        password1 = self.cleaned_data.get('password')
```
Comment (lhalam) [inline itaplay/itaplay/authentication/forms.py]: renam to password

File: itaplay/itaplay/authentication/forms.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        password2 = self.cleaned_data.get('confirm_password')
```
Comment (lhalam) [inline itaplay/itaplay/authentication/forms.py]: renam to password_confirm

File: itaplay/itaplay/authentication/views.py [hunk 3]
```diff
@@ -0,0 +1,179 @@
+import json
```
Comment (lhalam) [inline itaplay/itaplay/authentication/views.py:1]: grouped logically imports

File: .gitignore [hunk 4]
```diff
@@ -90,3 +90,9 @@ ENV/
 
 # Ignore for PyCharm settings
 .idea
+
+# Local DB
+db.sqlite3
+
+# PyLint config
+.pylintrc
```
Comment (lhalam) [inline .gitignore]: add newline at end of file

File: itaplay/itaplay/home/views.py [hunk 5]
```diff
@@ -3,7 +3,13 @@
 # Create your views here.
```
Comment (lhalam) [inline itaplay/itaplay/home/views.py]: delete comment

File: itaplay/itaplay/authentication/views.py [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+            return HttpResponse("incorect username or password", status=401)
+
+        #else:
+        return HttpResponse(status=400)
+
+
+    def get(self, request, *args, **kwargs):
+        return render(request, 'login.html')
+
+
```
Comment (lhalam) [inline itaplay/itaplay/authentication/views.py:145]: delete empty line

File: itaplay/itaplay/itaplay/urls.py [hunk 7]
```diff
@@ -17,7 +17,7 @@
 from django.contrib import admin
 
 urlpatterns = [
-    url(r'^', include('home.urls')),
+    url(r'^', include('home.urls')), #.*$
```
Comment (lhalam) [inline itaplay/itaplay/itaplay/urls.py]: delete comment

File: itaplay/itaplay/static/js/app/app.js [hunk 8]
```diff
@@ -1,16 +1,66 @@
...
+    $routeProvider
+    	.when('/test', {
             templateUrl: '../../../static/js/app/test/views/test.html',
             controller: TestController
-        }).
-        otherwise({redirectTo: '/test'});
-}).
-run(function($log) {
+        })
+        .otherwise({redirectTo: '/test'})
```
Comment (lhalam) [inline itaplay/itaplay/static/js/app/app.js]: delete this line, `otherwise` must by the only one

File: itaplay/itaplay/static/js/app/app.js [hunk 9]
```diff
@@ -1,16 +1,66 @@
...
+
+        .when('/test1', {
+            templateUrl: '../../../static/js/app/test/views/test1.html',
+
+        })
+        .otherwise({redirectTo: '/test'})
```
Comment (lhalam) [inline itaplay/itaplay/static/js/app/app.js]: delete this line, otherwise must by the only one

File: itaplay/itaplay/static/js/app/app.js [hunk 10]
```diff
@@ -1,16 +1,66 @@
...
+            method: 'POST',
+            url: $location.$$absUrl,    // can cause problems
+            data: {
+                first_name: $scope.registrationInfo.firstName,
+                last_name: $scope.registrationInfo.lastName,
+                password: $scope.registrationInfo.password,
+                confirm_password: $scope.registrationInfo.confirmPassword
+            }
+        };
+        console.log(req.data);
```
Comment (lhalam) [inline itaplay/itaplay/static/js/app/app.js]: delete this line

File: itaplay/itaplay/static/js/app/app.js [hunk 11]
```diff
@@ -1,16 +1,66 @@
...
+itaplay.config(['$httpProvider', function($httpProvider) {
+    $httpProvider.defaults.xsrfCookieName = 'csrftoken';
+    $httpProvider.defaults.xsrfHeaderName = 'X-CSRFToken';
+}]);
+
+itaplay.controller("RegisterCtrl",['$scope', '$http', '$location', '$window',
+    function ($scope, $http, $location, $window) {
+
+    $scope.registerUser = function() {
+        console.log("Start registration");
```
Comment (lhalam) [inline itaplay/itaplay/static/js/app/app.js]: delete this line

File: itaplay/itaplay/static/js/app/app.js [hunk 12]
```diff
@@ -1,16 +1,66 @@
...
+            data: {
+                first_name: $scope.registrationInfo.firstName,
+                last_name: $scope.registrationInfo.lastName,
+                password: $scope.registrationInfo.password,
+                confirm_password: $scope.registrationInfo.confirmPassword
+            }
+        };
+        console.log(req.data);
+        $http(req).success(function(){
+            console.log("Successful registration");
```
Comment (lhalam) [inline itaplay/itaplay/static/js/app/app.js]: delete this line

File: itaplay/itaplay/static/js/app/app.js [hunk 13]
```diff
@@ -1,16 +1,66 @@
...
+            $window.location.href = '/';
+        }).error(function(err){
+            alert(err);
```
Comment (lhalam) [inline itaplay/itaplay/static/js/app/app.js]: ? `alert` ?

File: itaplay/itaplay/static/js/app/app.js [hunk 14]
```diff
@@ -1,16 +1,66 @@
...
+        });
+    };
+}]);
```
Comment (lhalam) [inline itaplay/itaplay/static/js/app/app.js]: add newline at end of file

File: itaplay/itaplay/static/js/app/authentification/controllers/login-controller.js [hunk 15]
```diff
@@ -0,0 +1,15 @@
+angular.module('loginApp',['ngMessages'])
+.controller('LoginController', function($scope,$http, $window) {
+    $scope.LoginUser = function(user,password,form) {
+        console.log($scope);
```
Comment (lhalam) [inline itaplay/itaplay/static/js/app/authentification/controllers/login-controller.js]: delete this line

File: itaplay/itaplay/static/js/app/authentification/controllers/login-controller.js [hunk 16]
```diff
@@ -0,0 +1,15 @@
...
+        $http({
+            method: 'POST',
+            url: '/auth/login',
+            data: {username: user, password: password}
+        }).then(function successCallback(response) {
+            $window.location.href = '/';
+            },
+                function errorCallback(response) {console.log(response);});
```
Comment (lhalam) [inline itaplay/itaplay/static/js/app/authentification/controllers/login-controller.js]: update this code? not use `console.log`

File: itaplay/itaplay/static/js/app/test/views/test.html [hunk 17]
```diff
@@ -1 +1,7 @@
-<div>test1</div>
```
Comment (lhalam) [inline itaplay/itaplay/static/js/app/test/views/test.html]: ?

File: itaplay/itaplay/static/js/app/test/views/test.html [hunk 18]
```diff
@@ -1 +1,7 @@
...
\ No newline at end of file
+<div layout="row" class='coming_soon'>
+  <md-button flex="100">
+  <p>hello, geek</p>
+    <img src="/static/images/Logo.png">
+    <p>coming soon...</p>
+  </md-button>
+</div>
```
Comment (lhalam) [inline itaplay/itaplay/static/js/app/test/views/test.html]: add newline at end of file

File: itaplay/itaplay/static/js/app/views/footer.html [hunk 19]
```diff
@@ -0,0 +1,7 @@
+<footer>
+  <div layout="row" layout-align="center end" id="footer-content">
+    <p class="footer_text">&copy;AdViser</p>
+    <span flex></span>
+    <p class="footer_text">2016</p>
+  </div>
+</footer>
```
Comment (lhalam) [inline itaplay/itaplay/static/js/app/views/footer.html]: add newline at end of file

File: itaplay/itaplay/static/js/app/test/views/test1.html [hunk 20]
```diff
@@ -0,0 +1,8 @@
+<div layout="row" class='coming_soon'>
```
Comment (lhalam) [inline itaplay/itaplay/static/js/app/test/views/test1.html:1]: delete empty lines

File: itaplay/itaplay/static/js/app/test/views/login.html [hunk 21]
```diff
@@ -0,0 +1,36 @@
+<!DOCTYPE html>
```
Comment (lhalam) [inline itaplay/itaplay/static/js/app/test/views/login.html:1]: delete empty lines

File: itaplay/itaplay/static/js/app/test/views/clips.html [hunk 22]
```diff
@@ -0,0 +1,7 @@
+<div layout="row" class='coming_soon'>
+
```
Comment (lhalam) [inline itaplay/itaplay/static/js/app/test/views/clips.html]: delete empty line

File: itaplay/itaplay/static/js/app/views/header.html [hunk 23]
```diff
@@ -0,0 +1,26 @@
...
+</header> 
+
+<section layout="row" layout-align="center start">
+  <md-button href="#test" class="groupX left">users</md-button>
+  <md-button href="#test1" class="groupX middle">projects</md-button>
+  <md-button href="#clips" class="groupX middle">clips</md-button>
+  <md-button class="groupX middle">players</md-button>
+  <md-button class="groupX middle">messages</md-button>
+  <md-button class="groupX right">templates</md-button> 
+</section>
```
Comment (lhalam) [inline itaplay/itaplay/static/js/app/views/header.html]: add newline at end of file

File: itaplay/itaplay/static/styles/main.css [hunk 24]
```diff
@@ -0,0 +1,135 @@
...
+@-moz-keyframes inputHighlighter {
+  from { background:#5264AE; }
+  to  { width:0; background:transparent; }
+}
+@keyframes inputHighlighter {
+  from { background:#5264AE; }
+  to  { width:0; background:transparent; }
+}
+
+/* search form ending stylings ------------------------------- */
```
Comment (lhalam) [inline itaplay/itaplay/static/styles/main.css]: add newline at end of file

File: itaplay/itaplay/static/styles/register.css [hunk 25]
```diff
@@ -0,0 +1,7 @@
+.inputdemoErrors .inputErrorsApp {
+  min-height: 48px; }
```
Comment (lhalam) [inline itaplay/itaplay/static/styles/register.css]: move '}' in a new line

File: itaplay/itaplay/static/styles/register.css [hunk 26]
```diff
@@ -0,0 +1,7 @@
...
+.inputdemoErrors md-input-container > p {
+  font-size: 0.8em;
+  text-align: left;
+  width: 100%; }
```
Comment (lhalam) [inline itaplay/itaplay/static/styles/register.css]: move '}' in a new line

File: itaplay/itaplay/templates/index.html [hunk 27]
```diff
@@ -1,16 +1,35 @@
...
-<div ng-view></div>
 
+<div class="main_wrapper">
+  <div ng-include="'/static/js/app/views/header.html'"></div>
+	<div ng-view></div>
+</div>
+<ng-include src="'/static/js/app/views/footer.html'"></ng-include>
+
+
+	 <!-- Angular Material requires Angular.js Libraries -->
```
Comment (lhalam) [inline itaplay/itaplay/templates/index.html]: move js and css  include to `head`

File: itaplay/itaplay/templates/invite.html [hunk 28]
```diff
@@ -0,0 +1,33 @@
...
+  <h1>Invite user</h1>
+
+  <form action="" method="post">
+      {% csrf_token %}
+
+      {{ inviteForm.as_p }}
+      <input type="submit" value="Invite">
+  </form>
+	 <!-- Angular Material requires Angular.js Libraries -->
+<script src='//ajax.googleapis.com/ajax/libs/angularjs/1.5.8/angular.min.js'></script>
```
Comment (lhalam) [inline itaplay/itaplay/templates/invite.html]: move js and css  include to head

File: itaplay/itaplay/templates/invite.html [hunk 29]
```diff
@@ -0,0 +1,33 @@
...
+<script src="http://ajax.googleapis.com/ajax/libs/angularjs/1.5.3/angular-messages.min.js"></script>
+
+  <!-- Angular Material Library -->
+<script src="http://ajax.googleapis.com/ajax/libs/angular_material/1.1.0-rc2/angular-material.min.js"></script>
+
+  <!-- Our custom js -->
+<script src='/static/js/app/app.js'></script>
+<script src='/static/js/app/test/controllers/test-controller.js'></script>
+</body>
+</html>
```
Comment (lhalam) [inline itaplay/itaplay/templates/invite.html]: add newline at end of file

File: itaplay/itaplay/templates/invite.html [hunk 30]
```diff
@@ -0,0 +1,33 @@
+<!DOCTYPE html>
+<html ng-app="itaplay" >
+<head>
+	<title>Create an account</title>
```
Comment (lhalam) [inline itaplay/itaplay/templates/invite.html]: remove extra spaces and tabs

File: itaplay/itaplay/templates/login.html [hunk 31]
```diff
@@ -0,0 +1,45 @@
+<!DOCTYPE html>
```
Comment (lhalam) [inline itaplay/itaplay/templates/login.html:1]: delete empty lines

File: itaplay/itaplay/templates/register.html [hunk 32]
```diff
@@ -0,0 +1,36 @@
...
+  <form  ng-submit="registerUser()" ng-controller="RegisterCtrl">
+      <p><label>First name:</label> <input ng-model="registrationInfo.firstName"></p>
+      <p><label>Last name:</label> <input ng-model="registrationInfo.lastName"></p>
+      <p><label>Password:</label> <input type="password" ng-model="registrationInfo.password"></p>
+      <p><label>Confirm password:</label> <input type="password" ng-model="registrationInfo.confirmPassword"></p>
+      <input type="submit" value="Register user" >
+
+  </form>
+
+	 <!-- Angular Material requires Angular.js Libraries -->
```
Comment (lhalam) [inline itaplay/itaplay/templates/register.html]: move js and css  include to head

File: itaplay/itaplay/authentication/admin.py [hunk 33]
```diff
@@ -0,0 +1,3 @@
+from django.contrib import admin
```
Comment (ypyrko) [inline itaplay/itaplay/authentication/admin.py:1]: add models into django admin page.

File: itaplay/itaplay/authentication/forms.py [hunk 34]
```diff
@@ -0,0 +1,37 @@
+from models import AdviserInvitations
+from django import forms
+from django.contrib.auth.models import User
+
+class UserForm(forms.ModelForm):
```
Comment (ypyrko) [inline itaplay/itaplay/authentication/forms.py]: Rename form. Make it more specific. If this for using for registartion validation it should be named as UserRegistartionForm

File: itaplay/itaplay/authentication/forms.py [hunk 35]
```diff
@@ -0,0 +1,37 @@
...
+    """
+    Form for registration user
+    """
+    password = forms.CharField(widget=forms.PasswordInput, min_length=6, max_length=20)
+    confirm_password = forms.CharField(widget=forms.PasswordInput, label="Confirm password",
```
Comment (ypyrko) [inline itaplay/itaplay/authentication/forms.py:14]: Do we need only this fields for registration? I think we need more...

File: itaplay/itaplay/authentication/forms.py [hunk 36]
```diff
@@ -0,0 +1,37 @@
...
+
+    def clean_confirm_password(self):
+        """
+        Function checks that fields password and confirm_password are equal
+        :return: nothing
+        """
+        password1 = self.cleaned_data.get('password')
+        password2 = self.cleaned_data.get('confirm_password')
+
+        if not password2:
```
Comment (ypyrko) [inline itaplay/itaplay/authentication/forms.py]: You don't need this check. you can change the form to make this fields required forms.CharField(widget=forms.PasswordInput, min_length=6, max_length=20, requited=True) ...

File: itaplay/itaplay/authentication/forms.py [hunk 37]
```diff
@@ -0,0 +1,37 @@
...
+            raise forms.ValidationError("You must confirm your password")
+        if password1 != password2:
+            raise forms.ValidationError("Your passwords do not match")
+        return password2
+
+class InviteForm(forms.ModelForm):
```
Comment (ypyrko) [inline itaplay/itaplay/authentication/forms.py]: InviteForm -> UserInvitationForm

File: itaplay/itaplay/authentication/models.py [hunk 38]
```diff
@@ -0,0 +1,32 @@
...
+
+from django.db import models
+from django.contrib.auth.models import User
+
+class AdviserUser(models.Model):
+    """
+    Model of user - contain foreign key to default Django user and have additional fields
+    """
+    user = models.OneToOneField(User, on_delete=models.CASCADE)
+    ID_company = models.IntegerField() # will be foreign key
```
Comment (ypyrko) [inline itaplay/itaplay/authentication/models.py]: ID_company -> company_id

File: itaplay/itaplay/authentication/models.py [hunk 39]
```diff
@@ -0,0 +1,32 @@
...
+    avatar = models.URLField(default="default-user-logo.png")
+
+    def setup_user(self, base_user, invitation):
```
Comment (ypyrko) [inline itaplay/itaplay/authentication/models.py]: Why do we need this method (setup_user)?

File: itaplay/itaplay/authentication/models.py [hunk 40]
```diff
@@ -0,0 +1,32 @@
...
+        :return: nothing
+        """
+        self.user = base_user
+        self.ID_company = invitation.id_company
+
+
+class AdviserInvitations(models.Model):
+    """Stores invitation data"""
+    email = models.EmailField()
+    id_company = models.IntegerField()
```
Comment (ypyrko) [inline itaplay/itaplay/authentication/models.py]: Don't forget to change it to foreign key.

File: itaplay/itaplay/authentication/apps.py [hunk 41]
```diff
@@ -0,0 +1,7 @@
+from __future__ import unicode_literals
```
Comment (ypyrko) [inline itaplay/itaplay/authentication/apps.py]: Why do we need this file?

File: itaplay/itaplay/authentication/urls.py [hunk 42]
```diff
@@ -0,0 +1,9 @@
+from django.conf.urls import url
+from . import views
+
+
+urlpatterns = [
+    url(r'^register', views.RegistrationView.as_view()),
+    url(r'^invite', views.invite),
```
Comment (ypyrko) [inline itaplay/itaplay/authentication/urls.py]: change to class based views

File: itaplay/itaplay/authentication/views.py [hunk 43]
```diff
@@ -0,0 +1,156 @@
...
+        :param kwargs:
+        :return: BadRequest when verification code is incorrect or function in other case
+        """
+        verification_code = request.GET.get("code", "")
+
+        if verification_code:
+            if AdviserInvitations.objects.filter(verification_code=verification_code).exists():
+                invitation = AdviserInvitations.objects.get(verification_code=verification_code)
+
+                if not invitation.is_active:
```
Comment (ypyrko) [inline itaplay/itaplay/authentication/views.py:37]: Avoid many included block.

File: itaplay/itaplay/authentication/views.py [hunk 44]
```diff
@@ -0,0 +1,156 @@
...
+        :param self:
+        :param request:
+        :param args:
+        :param kwargs:
+        :return: BadRequest when verification code is incorrect or function in other case
+        """
+        verification_code = request.GET.get("code", "")
+
+        if verification_code:
+            if AdviserInvitations.objects.filter(verification_code=verification_code).exists():
```
Comment (ypyrko) [inline itaplay/itaplay/authentication/views.py]: You call 2 times the same query to DB. This is not good! AdviserInvitations.objects.filter(verification_code=verification_code) 

Change it to call once.

File: itaplay/itaplay/utils/EmailService.py [hunk 45]
```diff
@@ -0,0 +1,59 @@
+import uuid
+from django.utils import timezone
+from django.template import Context
+from django.core.mail import EmailMessage
+from django.template.loader import get_template
+from authentication.models import AdviserInvitations
+
+URL_REGISTRATION = "http://127.0.0.1:8000/auth/register?code="
```
Comment (lhalam) [inline itaplay/itaplay/utils/EmailService.py]: replace `http://127.0.0.1:8000/`

File: itaplay/itaplay/authentication/views.py [hunk 46]
```diff
@@ -0,0 +1,156 @@
...
+        invitation.used_time = timezone.now()
+        invitation.save()
+
+    def get_invitation(self, verification_code):  # may be move to Invitation model
+        """
+        Function for finding invitation by verification code
+        :param verification_code: verification code for user registration
+        :return: invitation object of Invitation Model
+        """
+        if AdviserInvitations.objects.filter(verification_code=verification_code).exists():
```
Comment (ypyrko) [inline itaplay/itaplay/authentication/views.py]: Code used two times. Move this as model method

File: itaplay/itaplay/authentication/views.py [hunk 47]
```diff
@@ -0,0 +1,156 @@
...
+    @validate_verification_code
+    def post(self, request):
+        """
+        Handling GET method
+        :param request: Request to View
+        :return: HttpResponse with code 201 if user is created or
+        HttpResponseBadRequest if request contain incorrect data
+        """
+        verification_code = request.GET.get("code", "")
+
```
Comment (ypyrko) [inline itaplay/itaplay/authentication/views.py:70]: why do you need empty line here?

File: itaplay/itaplay/authentication/views.py [hunk 48]
```diff
@@ -0,0 +1,156 @@
...
+        invitation = self.get_invitation(verification_code)
+
+        data = json.loads(request.body)
+        base_form = UserForm(data)
```
Comment (ypyrko) [inline itaplay/itaplay/authentication/views.py]: base_form -> user_form

File: itaplay/itaplay/authentication/views.py [hunk 49]
```diff
@@ -0,0 +1,156 @@
...
+
+        new_base_user = base_form.save(commit=False)
+        new_base_user.username = invitation.email
+        new_base_user.email = invitation.email
+        new_base_user.set_password(data['password'])
+        new_base_user.save()
+
+        new_extended_user = AdviserUser()
+        new_extended_user.setup_user(new_base_user, invitation)
+        new_extended_user.save()
```
Comment (ypyrko) [inline itaplay/itaplay/authentication/views.py]: Move code of user creating to model.

File: itaplay/itaplay/authentication/views.py [hunk 50]
```diff
@@ -0,0 +1,156 @@
...
+
+        self.close_invitation(invitation)
+
+        return HttpResponse(status=201)
+
+
+def invite(request):
```
Comment (ypyrko) [inline itaplay/itaplay/authentication/views.py]: class based views

File: itaplay/itaplay/authentication/views.py [hunk 51]
```diff
@@ -0,0 +1,156 @@
...
+    if request.method == 'POST':
+        invite_form = InviteForm(request.POST)
+        if not invite_form.is_valid():
+            return HttpResponseBadRequest("Invalid input data. Please edit and try again.")
+        if User.objects.filter(email=invite_form.data[u'email']).exists():
```
Comment (ypyrko) [inline itaplay/itaplay/authentication/views.py]: don't use defaul user model. We are working with AdviserInvitations

File: itaplay/itaplay/authentication/views.py [hunk 52]
```diff
@@ -0,0 +1,156 @@
...
+            return HttpResponseBadRequest("User with this e-mail is registered")
+        if AdviserInvitations.objects.filter(email=invite_form.data[u'email']).exists():
```
Comment (ypyrko) [inline itaplay/itaplay/authentication/views.py:106]: The same code as above

File: itaplay/itaplay/authentication/views.py (hunk 50 above)
Comment (ypyrko) [inline itaplay/itaplay/authentication/views.py]: This should be available ONLY for logged in user (admin user)

File: itaplay/itaplay/authentication/views.py [hunk 53]
```diff
@@ -0,0 +1,156 @@
...
+
+    return render(request, "invite.html", {
+        'inviteForm': invite_form
+    })
+
+
+class LoginView(View):
+
+    def post(self, request):
+        data = json.loads(request.body)
```
Comment (ypyrko) [inline itaplay/itaplay/authentication/views.py:125]: Where is form validation here?

File: itaplay/itaplay/authentication/views.py [hunk 54]
```diff
@@ -0,0 +1,156 @@
...
+        password = data.get('password', None)
+
+        user = auth.authenticate(username=username, password=password)
+
+        if user is not None:
+            auth.login(request, user)
+            return HttpResponse(status=200)
+
+        else:
+            return HttpResponse("incorrect username or password", status=401)
```
Comment (ypyrko) [inline itaplay/itaplay/authentication/views.py]: HttpResponse -> HttpBadRequest (check the correct name)

File: itaplay/itaplay/itaplay/settings.py [hunk 55]
```diff
@@ -37,13 +47,16 @@
...
+    'authentication',
+    'utils',
 ]
 
 MIDDLEWARE_CLASSES = [
     'django.middleware.security.SecurityMiddleware',
     'django.contrib.sessions.middleware.SessionMiddleware',
     'django.middleware.common.CommonMiddleware',
-    'django.middleware.csrf.CsrfViewMiddleware',
+    # 'django.middleware.csrf.CsrfViewMiddleware',
```
Comment (ypyrko) [inline itaplay/itaplay/itaplay/settings.py]: uncomment this

File: itaplay/itaplay/authentication/forms.py [hunk 56]
```diff
@@ -0,0 +1,39 @@
+from models import AdviserInvitations
+from django import forms
+from django.contrib.auth.models import User
+
+
+class UserRegistrationForm(forms.ModelForm):
+    """
+    Form for registration user
+    """
```
Comment (ypyrko) [inline itaplay/itaplay/authentication/forms.py:9]: I think we need more fields for validation here

File: itaplay/itaplay/authentication/models.py [hunk 57]
```diff
@@ -0,0 +1,61 @@
...
+        self.save()
+
+    @staticmethod
+    def get_invitation(verification_code):
+        """
+        Function for finding invitation by verification code
+        :param verification_code: verification code for user registration
+        :return: invitation object of Invitation Model
+        """
+        invitation_query = AdviserInvitations.objects.filter(verification_code=verification_code)
```
Comment (ypyrko) [inline itaplay/itaplay/authentication/models.py:61]: Think about this code. Rebuild it

File: itaplay/itaplay/authentication/models.py [hunk 58]
```diff
@@ -0,0 +1,61 @@
...
+        if len(invitation_query):
```
Comment (ypyrko) [inline itaplay/itaplay/authentication/models.py]: invitation_query.exists()

File: itaplay/itaplay/authentication/models.py [hunk 59]
```diff
@@ -0,0 +1,61 @@
...
+            invitation = invitation_query[0]
```
Comment (ypyrko) [inline itaplay/itaplay/authentication/models.py]: invitation_query.firsts()

File: itaplay/itaplay/authentication/views.py [hunk 60]
```diff
@@ -0,0 +1,151 @@
...
+        :param request: Request to View
+        :return: HttpResponse with code 201 if user is created or
+        HttpResponseBadRequest if request contain incorrect data
+        """
+        verification_code = request.GET.get("code", "")
+
+        invitation = AdviserInvitations.get_invitation(verification_code)
+
+        data = json.loads(request.body)
+        user_registration_form = UserRegistrationForm(data)
```
Comment (ypyrko) [inline itaplay/itaplay/authentication/views.py]: UserRegistrationForm(data=json.loads(request.body))

File: itaplay/itaplay/authentication/views.py [hunk 61]
```diff
@@ -0,0 +1,151 @@
...
+        Handling GET method
+        :param request: Request to View
+        :return: HttpResponse with code 201 if user is created or
+        HttpResponseBadRequest if request contain incorrect data
+        """
+        verification_code = request.GET.get("code", "")
+
+        invitation = AdviserInvitations.get_invitation(verification_code)
+
+        data = json.loads(request.body)
```
Comment (ypyrko) [inline itaplay/itaplay/authentication/views.py]: drop this line

File: itaplay/itaplay/authentication/views.py [hunk 62]
```diff
@@ -0,0 +1,151 @@
...
+        user_registration_form = UserRegistrationForm(data)
+
+        if not user_registration_form.is_valid():
+            return HttpResponseBadRequest("Invalid input data. Please edit and try again.")
+
+        new_user = AdviserUser(user_registration_form, invitation)
+        new_user.save()
```
Comment (ypyrko) [inline itaplay/itaplay/authentication/views.py]: check if we need this line

File: itaplay/itaplay/authentication/views.py [hunk 63]
```diff
@@ -0,0 +1,151 @@
...
+            :return: HttpResponse with code 201 if user is invited or
+                     HttpResponseBadRequest if request contain incorrect data
+        """
+
+        invite_form = UserInvitationForm(request.POST)
+
+        if not invite_form.is_valid():
+            return HttpResponseBadRequest("Invalid input data. Please edit and try again.")
+
+        if User.objects.filter(email=invite_form.data[u'email']).exists():
```
Comment (ypyrko) [inline itaplay/itaplay/authentication/views.py]: We are not working with default user. We can do it using AdvisorUser model

File: itaplay/itaplay/authentication/views.py [hunk 64]
```diff
@@ -0,0 +1,151 @@
...
+            return HttpResponseBadRequest("User with this e-mail is registered")
+
+        if AdviserInvitations.objects.filter(email=invite_form.data[u'email']).exists():
+            return HttpResponseBadRequest("User with this e-mail is already invited")
+
+        sender = EmailSender(invite_form.data[u'email'])
+        print sender.send_invite(invite_form.data[u'id_company'])
```
Comment (ypyrko) [inline itaplay/itaplay/authentication/views.py]: Drop print. Use logger here instead of prints

File: itaplay/itaplay/static/js/app/app.js [hunk 65]
```diff
@@ -1,16 +1,82 @@
...
+      .primaryPalette('teal')
+      .accentPalette('blue');
+});
+
+itaplay.config(['$httpProvider', function($httpProvider) {
+    $httpProvider.defaults.xsrfCookieName = 'csrftoken';
+    $httpProvider.defaults.xsrfHeaderName = 'X-CSRFToken';
+}]);
+
+itaplay.controller("RegisterCtrl",['$scope', '$http', '$location', '$window', '$mdDialog',
```
Comment (ypyrko) [inline itaplay/itaplay/static/js/app/app.js]: Move to separate file

File: itaplay/itaplay/static/js/app/app.js [hunk 66]
```diff
@@ -1,16 +1,82 @@
...
+                    .clickOutsideToClose(true)
+                    .title('Error in form')
+                    .textContent(err)
+                    .ok('Ok')
+            );
+        });
+    };
+}]);
+
+itaplay.directive("compareTo", function(){
```
Comment (ypyrko) [inline itaplay/itaplay/static/js/app/app.js]: move to separate file.

File: itaplay/itaplay/static/js/app/authentification/controllers/login-controller.js [hunk 67]
```diff
@@ -0,0 +1,13 @@
+angular.module('loginApp',['ngMessages'])
```
Comment (ypyrko) [inline itaplay/itaplay/static/js/app/authentification/controllers/login-controller.js]: check spaces here

File: itaplay/itaplay/authentication/views.py (hunk 63 above)
Comment (yurkovoznyak) [inline itaplay/itaplay/authentication/views.py]: Done

File: itaplay/itaplay/authentication/views.py (hunk 64 above)
Comment (yurkovoznyak) [inline itaplay/itaplay/authentication/views.py]: Done

File: itaplay/itaplay/authentication/forms.py (hunk 56 above)
Comment (vova-kul) [inline itaplay/itaplay/authentication/forms.py:9]: Done

File: itaplay/itaplay/authentication/models.py (hunk 57 above)
Comment (vova-kul) [inline itaplay/itaplay/authentication/models.py:61]: Done

File: itaplay/itaplay/authentication/models.py (hunk 58 above)
Comment (vova-kul) [inline itaplay/itaplay/authentication/models.py]: Done

File: itaplay/itaplay/authentication/models.py (hunk 59 above)
Comment (vova-kul) [inline itaplay/itaplay/authentication/models.py]: Done

File: itaplay/itaplay/authentication/views.py (hunk 60 above)
Comment (vova-kul) [inline itaplay/itaplay/authentication/views.py]: Done

File: itaplay/itaplay/authentication/views.py (hunk 61 above)
Comment (vova-kul) [inline itaplay/itaplay/authentication/views.py]: Done

File: itaplay/itaplay/authentication/views.py (hunk 62 above)
Comment (vova-kul) [inline itaplay/itaplay/authentication/views.py]: Done

File: itaplay/itaplay/static/js/app/app.js (hunk 65 above)
Comment (vova-kul) [inline itaplay/itaplay/static/js/app/app.js]: Done

File: itaplay/itaplay/static/js/app/app.js (hunk 66 above)
Comment (vova-kul) [inline itaplay/itaplay/static/js/app/app.js]: Done

**Pull request merged**